File: src/main/java/dev/obscuria/tooltips/client/tooltip/filter/NeverFilter.java
package dev.obscuria.tooltips.client.tooltip.filter;

import com.github.bsideup.jabel.Desugar;
import net.minecraft.item.ItemStack;

@Desugar
public record NeverFilter() implements ItemFilter {

    public static final NeverFilter INSTANCE = new NeverFilter();

    @Override
    public boolean test(ItemStack stack) {
        return false;
    }
}

File: common/src/main/kotlin/dev/obscuria/tooltips/client/filter/NeverFilter.kt
package dev.obscuria.tooltips.client.filter

import com.mojang.serialization.Codec
import net.minecraft.world.item.ItemStack

object NeverFilter : IItemFilter {

    val CODEC: Codec<NeverFilter> = Codec.unit(NeverFilter)

    override fun codec(): Codec<NeverFilter> = CODEC

    override fun test(stack: ItemStack): Boolean = false
}